coordinator/src/collaborative_revert.rs

```diff
@@ -28,254 +29,242 @@ use dlc_manager::Storage;
-        .context("Could not get channel")?;
+    let channel_id_hex = channel_id.to_hex();
 
-    let sub_channels = node
+    let (ln_inbound_liquidity_sat, ln_outbound_liquidity_sat) =
+        match node.channel_manager.get_channel_details(&channel_id) {
+            Some(details) => {
+                let ln_inbound_liquidity_sat =
+                    Decimal::from(details.inbound_capacity_msat) / Decimal::ONE_THOUSAND;
+                let ln_inbound_liquidity_sat = ln_inbound_liquidity_sat
+                    .to_u64()
+                    .expect("inbound liquidity to fit into u64");
+
+                let ln_outbound_liquidity_sat =
+                    Decimal::from(details.outbound_capacity_msat) / Decimal::ONE_THOUSAND;
+                let ln_outbound_liquidity_sat = ln_outbound_liquidity_sat
+                    .to_u64()
+                    .expect("outbound liquidity to fit into u64");
+
+                (ln_inbound_liquidity_sat, ln_outbound_liquidity_sat)
+            }
+            None => {
+                let ln_liquidity = ln_liquidity
+                    .context("Unknown Lightning liquidity. Must provide that information")?;
+
+                (
+                    ln_liquidity.inbound_liquidity,
+                    ln_liquidity.outbound_liquidity,
+                )
+            }
+        };
+
+    let subchannels = node
         .list_dlc_channels()
-        .context("Could not list dlc channels")?;
+        .context("Could not get list of subchannels")?;
 
-    let sub_channel = sub_channels
+    let subchannel = subchannels
         .iter()
         .find(|c| c.channel_id == channel_id)
-        .context("Could not find provided channel")?;
+        .context("Missing subchannel")?;
 
-    let position =
-        Position::get_position_by_trader(&mut conn, channel_details.counterparty.node_id, vec![])?
-            .context("Could not load position for channel_id")?;
+    let peer_id = subchannel.counter_party;
+    let fund_txo_sat = subchannel.fund_value_satoshis;
+
+    // The coordinator's unspendable reserve is almost certainly 1% of the funding TXO, although
+    // this depends on the `their_channel_reserve_proportional_millionths` config value set by the
+    // counterparty.
+    let unspendable_punishment_reserve_sat = {
+        let fund_txo_sat = Decimal::from(fund_txo_sat);
+
+        let reserve = fund_txo_sat / Decimal::ONE_HUNDRED;
 
-    let settlement_amount = position
-        .calculate_settlement_amount(revert_params.price)
-        .context("Could not calculate settlement amount")?;
+        // The minimum reserve is _at least_ 1_000 sats.
+        let reserve = reserve.max(Decimal::ONE_THOUSAND);
 
-    let trade_settled = sub_channel.fund_value_satoshis == channel_details.channel_value_satoshis;
+        reserve.to_u64().expect("Reserve to fit into u64")
+    };
+
+    let coordinator_amount = if subchannel.state == SubChannelState::OffChainClosed {
```

Comment:
🔧 I don't think that this substitutes this `let trade_settled = sub_channel.fund_value_satoshis == channel_details.channel_value_satoshis;`

I added this `trade_settled` flag because it could happen that the lightning channel balance had already been updated even if the dlc protocol is still stuck (this is typically the reason why we want to collab revert).

Just checking if the subchannel state is already `OffChainClosed` is (as far as I remember) not enough.

Reply:
Okay, I'll look into this. The name `trade_settled` was misleading.

Reply:
I've decided to use the `ChannelDetails` here if available and let the caller pass in an `is_split` flag (together with the liquidity information) otherwise.

Given that this is an API that we must use with caution I think it's reasonable to expect us to provide these key values if everything else fails.